e2e/test/PlugAndPlayTests.cs

```diff
@@ -0,0 +1,326 @@
+                        Pooling = true
+                    }
+                }
+            };
+
+            // With PoolingOverAmqp.MultipleConnections_DevicesCount = 4, and PoolingOverAmqp.MultipleConnections_PoolSize = 2:
+            // Create a total of 4 device client instances, multiplexed over 2 amqp connections. Each device client instance uses a different ModelId.
+            string modelIdPrefix = "dtmi:com:example:testModel";
+            for (int i = 0; i < PoolingOverAmqp.MultipleConnections_DevicesCount; i++)
+            {
+                TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix, TestDeviceType.Sasl).ConfigureAwait(false);
+
+                string modelId = $"{modelIdPrefix};{testDevice.Id}";
+                DeviceClient deviceClient = testDevice.CreateDeviceClient(transportSettings, authScope, new ClientOptions { ModelId = modelId });
+                devices.Add(testDevice, deviceClient);
+            }
+
+            // Calling OpenAsync() will cause the client to open the connection.
+            foreach (DeviceClient client in devices.Values)
+            {
+                operations.Add(client.OpenAsync());
+            }
+            await Task.WhenAll(operations).ConfigureAwait(false);
+
+            // Act and Assert
+
+            // Retrieve the device twin and assert the value of ModelId returned.
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            foreach (TestDevice device in devices.Keys)
+            {
+                string deviceId = device.Id;
+                Twin twin = await registryManager.GetTwinAsync(deviceId).ConfigureAwait(false);
+                twin.ModelId.Should().Be($"{modelIdPrefix};{deviceId}", "because the device was created as plug and play");
+            }
+
+            // Cleanup
+            foreach (TestDevice device in devices.Keys)
+            {
+                await registryManager.RemoveDeviceAsync(device.Id).ConfigureAwait(false);
+            }
+        }
+
+        private static async Task CreateMultiplexedModulesAndRetrieveTwinAsync(Client.TransportType transportType)
+        {
+            // Setup
+            var modules = new Dictionary<TestModule, ModuleClient>();
+            var operations = new List<Task>();
+
+            var transportSettings = new ITransportSettings[]
+            {
+                new AmqpTransportSettings(transportType)
+                {
+                    AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                    {
+                        MaxPoolSize = unchecked((uint)PoolingOverAmqp.MultipleConnections_PoolSize),
+                        Pooling = true
+                    }
+                }
+            };
+
+            // With PoolingOverAmqp.MultipleConnections_DevicesCount = 4, and PoolingOverAmqp.MultipleConnections_PoolSize = 2:
```

Comment:
I am not sure about this comment, what if those constants change? if we want to test with specific values shouldn't we just not use constants here?

Reply:
There was no way for me to reference these constant values in these dev comments. The public summary comments can reference them directly by using <see cref=""/>, but that doesn't work for inline dev comments.
If these values change, the tests will still be valid, they'll just be tested against a different batch size.

The same logic is used for other multiplexing/ pooling tests as well.